.github/workflows/tests.yaml

```diff
@@ -312,32 +312,35 @@ jobs:
         run: nox -s fragment-${{ matrix.python }}
   integration:
     runs-on: ubuntu-latest
+    container: gcr.io/gapic-images/googleapis-bazel:20210105
     steps:
       - name: Cancel Previous Runs
         uses: styfle/cancel-workflow-action@0.7.0
         with:
           access_token: ${{ github.token }}
       - uses: actions/checkout@v2
-      - name: Set up Python 3.8
-        uses: actions/setup-python@v2
+      - name: Cache Bazel files
+        id: cache-bazel
+        uses: actions/cache@v2
         with:
-          python-version: 3.8
-      - name: Install system dependencies.
+          path: ~/.cache/bazel
+          key: ${{ runner.os }}-bazel-20210105-${{ secrets.CACHE_VERSION }}
+      - name: Cache not found
+        if: steps.cache-bazel.outputs.cache-hit != 'true'
         run: |
-          sudo apt-get update
-          sudo apt-get install -y curl pandoc unzip gcc
-      - name: Install Bazel
+          echo "No cache found."
```

Comment:
Shouldn't a cache miss imply that we actually do something to populate the missed files?

Reply:
This is done implicitly by bazel during startup. It checks for the existence of `~/.cache/bazel` and if not found runs all the tedious startup of compiling `gRPC`, `protoc`, etc. which are then put into the cache.
The github action saves this directory persistently. At the start of a CI run, it checks for that directory in persistent storage and, if found, populates it in the container.

Reply:
Anecdotally, this cuts the integration test down to about 1 minute 30 seconds.